perspective_reddit_bot/check_mod_actions.py

```diff
@@ -0,0 +1,110 @@
+import json
+import praw
+import time
+
+from creds import creds
+
+
+def write_moderator_actions(reddit,
+                            line,
+                            id_key,
+                            timestamp_key,
+                            output_path,
+                            hours_to_wait):
+  record = json.loads(line)
+  comment = reddit.comment(record[id_key])
+  maybe_wait(record[timestamp_key], hours_to_wait)
+  record['approved'] = comment.approved
+  record['removed'] = comment.removed
+  with open(output_path, 'a') as o:
+    json.dump(record, o)
+    o.write('\n')
+
+
+def maybe_wait(timestamp, hours_to_wait):
+  """Waits until hours_to_wait hours have passed since timestamp"""
+
+  now = datetime.utcnow()
+  time_diff = now - datetime.strptime(timestamp, '%Y%m%d_%H%M%S')
+  time_diff = time_diff.seconds
+  seconds_to_wait = hours_to_wait * 3600
+  if time_diff < seconds_to_wait:
+    time_to_wait = seconds_to_wait - time_diff
+    print('Waiting %.1f seconds...' % time_to_wait)
+    time.sleep(time_to_wait)
+
+
+def _main():
+  parser = argparse.ArgumentParser(
+      'Reads the output of moderate_subreddit.py and adds actions taken by'
+      'human moderators.')
+  parser.add_argument('input_path', help='json file with reddit comment ids')
+  parser.add_argument('output_path', help='path to write output file')
+  parser.add_argument('-id_key', help='json key containing reddit comment id',
+                      default='comment_id')
+  parser.add_argument('-timestamp_key', help='json key containing timestamp'
+                      'that moderation bot saw comment',
+                      default='bot_review_utc')
+  parser.add_argument('-hours_to_wait',
+                      help='the number of hours to wait to allow moderators to'
+                      ' respond to bot',
+                      type=int,
+                      default=12)
+  parser.add_argument('-stop_at_eof',
+                      help='if set, stops the process once the end of file is '
+                      'hit instead of waiting for new comments to be written',
+                      action='store_true')
+
+  args = parser.parse_args()
+
+  reddit = praw.Reddit(client_id=creds['reddit_client_id'],
+                       client_secret=creds['reddit_client_secret'],
+                       user_agent=creds['reddit_user_agent'],
+                       username=creds['reddit_username'],
+                       password=creds['reddit_password'])
+
+  with open(args.input_path) as f:
+    # Loops through the file and waits at EOF for new data to be written.
+    while True:
+      where = f.tell()
+      line = f.readline()
+      if line:
+        write_moderator_actions(reddit,
+                                line,
+                                args.id_key,
+                                args.timestamp_key,
+                                args.output_path,
+                                args.hours_to_wait)
+      else:
```

Comment:
I think the else block here can be out-dented?

```
if line:
  write_moderator_actions(...)
else if args.stop_at_eof:
  return
else:
  print(...)
  time.sleep(...)
  f.seek(where)

Reply:
Done.